### PR TITLE
[RSDK-2905] Add pin definitions for Jetson Orin Nano

### PR DESCRIPTION
**Pull request by penguinland**
I haven't tried this out yet (will do that next time I'm in the office), but I strongly suspect this is ready to go. The Orin NX and Orin Nano have identical pin mappings, and the Orin NX's mappings are already in our code. I saw 3 approaches I could have taken:
- copy and paste the NX mappings. It's straightforward, but hides the information that they're identical, and if we ever change one (e.g., NVidia's official mapping for the Orin AGX was missing some pins which we added in https://github.com/viamrobotics/rdk/pull/2181), we'd need to remember to do it twice. Not a good choice.
- refactor the file to define a bunch of pin mappings, and then make a map from boards to mappings. Easiest to read because every variable will be constant, but requires pretty major rework from the file's current state (in which we make the large map all at once). If I had to rewrite the file from scratch, this is the approach I'd prefer.
- make the large map the way we currently do it, and then modify it to add one extra key-value pair whose value is taken from elsewhere in the map. This is the smallest change from the previous state of the file, but it could be surprising to people who don't realize we're about to mutate the map we _just_ created.

I had gone with the third approach because it was smallest/simplest, and then while writing this PR description convinced myself that the second approach is best, so I redid the PR. If you prefer the third approach, we can revert all but the first commit in this PR. I suspect that reviewing this file while ignoring whitespace changes will make things significantly simpler.

(Also, while I was at it, I renamed the Jetson Orin to it's full name, Jetson Orin AGX, to distinguish it from the Jetson Orin NX and Jetson Orin Nano.)

### CODE REVIEW

**Comment by penguinland** [comment]
The bucket randomizer gave me someone who mostly works in Rust these days. Nicolas, if you prefer, I can find someone else to review this instead.

**Comment by npmenard** [comment]
This looks good to me. I like #2 better too, it's more readable.

**Review by npmenard — APPROVED** [review]

File: components/board/jetson/data.go [hunk 1]
```diff
@@ -12,107 +12,387 @@ const (
...
+	{map[int]int{40: 0, 30: 0}, map[int]string{30: "PAA.00"}, "c2f0000.gpio", 33, 13, "CAN1_DOUT", "CAN1_DOUT", "", -1},
+	{map[int]int{224: 66, 169: 54}, map[int]string{169: "PI.02"}, "2200000.gpio", 35, 19, "I2S2_FS", "DAP2_FS", "", -1},
+	// Input-only (due to base board)
+	{map[int]int{224: 141, 169: 113}, map[int]string{169: "PR.05"}, "2200000.gpio", 36, 16, "UART1_CTS", "UART1_CTS", "", -1},
+	{map[int]int{40: 1, 30: 1}, map[int]string{30: "PAA.01"}, "c2f0000.gpio", 37, 26, "CAN1_DIN", "CAN1_DIN", "", -1},
+	{map[int]int{224: 65, 169: 53}, map[int]string{169: "PI.01"}, "2200000.gpio", 38, 20, "I2S2_DIN", "DAP2_DIN", "", -1},
+	{map[int]int{224: 64, 169: 52}, map[int]string{169: "PI.00"}, "2200000.gpio", 40, 21, "I2S2_DOUT", "DAP2_DOUT", "", -1},
+}
+
+//nolint:dupl // This is not actually a duplicate of jetsonTX2NXPins despite what the linter thinks
```
[the remark this answers is not in the document]
Comment (penguinland) [reply components/board/jetson/data.go:68]: Yeah, I got surprised! I think what's going on is that, like, 80% of the data is the same (pin names, etc.), so the linter says surely there's some way to reuse that without having it twice. but the 20% that's different (which line on the chip goes with which pin) is far more important.

**Comment by penguinland** [comment]
Tried on a Jetson Orin Nano: things seem to work alright (though like with https://github.com/viamrobotics/rdk/pull/2181, I suspect some pin definitions are missing. but those can be added in later). I also tried it on the Jetson Orin AGX: none of the old functionality has broken.

**Pull request merged**